malariagen_data/anoph/dipclust.py

```diff
@@ -456,28 +468,39 @@ def concat_subplots(
             plot_bgcolor="white",
         )
 
+        # COMMENT: this feels overly complex - can we somehow remove the colorbar earlier,
+        # when the heatmap traces are being created?
+        #
         # remove colorbar for all heatmap traces. This has to be determined dynamically as the
         # number of traces can vary depending on scatter color/symbol variables
-        for trace in [trace for trace in fig["data"] if trace.type == "heatmap"]:
-            idx = int(trace["yaxis"].lstrip("y"))
-            fig.update_traces(showscale=False, row=idx, col=0)
+        for trace in fig["data"]:
+            if trace.type == "heatmap":
+                idx = int(trace["yaxis"].lstrip("y"))
+                fig.update_traces(showscale=False, row=idx, col=0)
 
         return fig
 
     @doc(
-        summary="Perform diplotype clustering with heterozygosity and amino acid variants",
+        summary="Perform diplotype clustering, annotated with heterozygosity, gene copy number and amino acid variants.",
         parameters=dict(
             heterozygosity="Plot heterozygosity track.",
-            amino_acids="Plot amino acid variants.",
+            aa_transcript="Plot amino acid variants for this transcript.",
+            cnv_region="Plot gene CNV calls for this region.",
             leaf_y="Y coordinate at which to plot the leaf markers.",
-            filter_min_maf="Filter amino acid variants with alternate allele frequency below this threshold.",
-            cnv_region="The region to plot CNV calls for.",
+            aa_filter_min_maf="Filter amino acid variants with alternate allele frequency below this threshold.",
         ),
     )
     def plot_diplotype_clustering_advanced(
         self,
         region: base_params.regions,
-        transcript: Optional[base_params.transcript] = None,
+        heterozygosity: bool = True,
+        heterozygosity_colorscale: plotly_params.color_continuous_scale = "Greys",
+        aa_transcript: Optional[base_params.transcript] = None,
```

Comment:
Suggestion, instead of two parameters `amino_acids` and `transcript`, just have a single parameter `aa_transcript`.